py/visdom/__init__.py

```diff
@@ -796,7 +802,7 @@ def get_env_list(self):
         This function returns a list of all of the env names that are currently
         in the server.
         """
-        return json.loads(self._send({}, endpoint='env_state', quiet=True))
+        return list(self.env_list)
```

Comment:
Super close this time! The only thing is that when the server is running (rather than in offline mode) it is possible to change the envs from the server side. As such, you should only be returning this list when in offline mode.